server/src/main/java/io/crate/expression/scalar/HasPrivilegeFunction.java

```diff
@@ -218,4 +213,21 @@ protected static void validateCallPrivileges(Roles roles, Role sessionUser, Role
             throw new MissingPrivilegeException(sessionUser.name());
         }
     }
+
+    public interface CheckPrivilege {
+        Boolean check(Roles roles, Role user, Object object, Collection<Permission> permissions, Schemas schemas);
+    }
+
+    public interface ParsePermissions {
+
+        /**
+         * @param permissionNames is a comma separated list.
+         * Valid permissionNames are 'CREATE' and 'USAGE' which map to DDL and DQL correspondingly.
```

Comment:
This is not correct as for each function the `permissionNames` differ. we need to explain those on the concrete implementation and here have a more generic comment.